coordinator/core/core.go

```diff
@@ -79,6 +80,18 @@ const (
 	sKCoordinatorIntermediateKey  string = "coordinatorIntermediateKey"
 )
 
+// components is a list of the objects stored in the Coordinators store
```

Comment:
The idea of components is essentially a cache that simplifies accessing all values of a certain type, right?
For a single instance Coordinator, this is fine. If we go down the route of replication this structure is essentially useless.
Not sure if it makes sense to introduce it now with replication as a feature on the horizon.

Reply:
Correct, otherwise we would need to load the manifest, go through it to learn what values exists, and then request them. This is essentially what the core does at the moment when loading users or secrets.

Regarding replication: As far as I understand it, at no point in time do we want to create new objects that are not documented in the manifest. This means that, once we have a manifest, we can generate a list of all components that will exists throughout the coordinators lifetime.
A newly duplicated coordinator instance has access to the same store, so, upon setup, the instance would request the manifest, and is then able to generate the same cache of component as any other instance of the coordinator.

Reply:
Moritz is right. And it's not only about replication, but also about code complexity and maintainability.

> otherwise we would need to load the manifest, go through it to learn what values exists

The idiomatic way is iterating over the keys of a KV store.

> at no point in time do we want to create new objects that are not documented in the manifest

I bet we will do this one day ;)

That said, the components thing is fine with me as an intermediate stage. But the long-term goal should be that the store is the single source of truth. Once we have this, new things will be easier to implement. We've already seen complexity reductions of existing features in the previous PRs and also in this PR.